iot-hub/Samples/service/DigitalTwinClientSamples/Thermostat/ThermostatSample.cs

```diff
@@ -22,7 +22,7 @@ public ThermostatSample(DigitalTwinClient client, string digitalTwinId, ILogger
         {
             _digitalTwinClient = client ?? throw new ArgumentNullException(nameof(client));
             _digitalTwinId = digitalTwinId ?? throw new ArgumentNullException(nameof(digitalTwinId));
-            _logger = logger ?? LoggerFactory.Create(builder => builder.AddConsole()).CreateLogger<ThermostatSample>();
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
```

Comment:
I notice this is similar to one of changes Nate brought in another [PR](https://github.com/Azure-Samples/azure-iot-samples-csharp/pull/302#discussion_r945893919), so I'd like to ask same question here.

Reply:
I believe the reason is because `_logger` is an IDisposable and should be disposed of before references to it are out of scope. The specific change to throw an ArgumentNullException instead of creating a default logger was suggested by David to Nate. It seems like it could be for consistency with the `client` and `digitalTwinId`?